sources/ghostdog/src/main.rs

```diff
@@ -125,6 +241,46 @@ fn parse_device_name(device_info: &[u8], path: String) -> Result<String> {
         .to_string())
 }
 
+/// Read a file into a SupportedDevicesConfiguration Enum
+fn read_supported_devices_file(path: PathBuf) -> Result<SupportedDevicesConfiguration> {
+    let mut supported_devices_file =
+        fs::File::open(&path).context(error::OpenFileSnafu { path: path.clone() })?;
+    let mut supported_devices_str = String::new();
+    supported_devices_file
+        .read_to_string(&mut supported_devices_str)
+        .context(error::ReadFileSnafu { path: path.clone() })?;
+    let device_configuration: SupportedDevicesConfiguration =
+        serde_json::from_str(supported_devices_str.as_str())
+            .context(error::ParseGpuDevicesFileSnafu {})?;
+    Ok(device_configuration)
+}
+
+/// Given a PCI ID, search the Open GPU Supported Devices File to determine if the Open GPU Driver should be used
+fn find_preferred_driver(pci_id: String) -> Result<String> {
+    // This corresponds to T4 devices which are Turing based and supported, but currently do not show up in the supported-devices file
+    if pci_id == "10DE:1EB8" {
```

Comment:
I noticed that the g5g instances also don't match up on the supported list https://github.com/NVIDIA/open-gpu-kernel-modules?tab=readme-ov-file#compatible-gpus but are the Turing generation of devices that should support it. I'll do some more testing to see if the driver works on these instances too.

Reply:
I was able to get light confirmation from NVIDIA that these devices should use the proprietary driver so I've removed this if statement.